tests/test_deprecations.py

```diff
@@ -141,9 +141,13 @@ def test_action(
             "--foo",
             action=deprecated.action("2.0", "3.0", _StoreTrueAction),
         )
+        parser.add_argument(
+            "bar",
+            action=deprecated.action("2.0", "3.0", _StoreAction),
+        )
 
         with pytest.warns(warning, match=message):
-            parser.parse_args(["--foo"])
+            parser.parse_args(["--foo", "bar"])
```

Comment:
This will only test the `--foo` warning so we need to add a second case:

```suggestion
            parser.parse_args(["--foo"])

        with pytest.warns(warning, match=message):
            parser.parse_args(["bar"])
```

Reply:
The `--foo` flag passed without a value won't trigger a deprecation warning since the warning only comes up for marked flags that contain a value, thus the `"some_value"` addition to that line.